python/ray/worker.py

```diff
@@ -488,7 +488,7 @@ def init(
         _enable_object_reconstruction=False,
         _redis_max_memory=None,
         _plasma_directory=None,
-        _node_ip_address=ray_constants.NODE_DEFAULT_IP,
+        _node_ip_address=None,
```

Comment:
Why change this?

Also, can you make sure the original lines from the old code (https://github.com/ray-project/ray/pull/10325/files) are preserved?
```
    # Convert hostnames to numerical IP address.	
    if node_ip_address is not None:	
        node_ip_address = services.address_to_ip(node_ip_address)
```

Reply:
`_node_ip_address` was meaningless arg before because we didn't use it. 

If I understood correctly, this is supposed to be passed to `RayParam`'s `node_ip_address`. `RayParam.node_ip_address` is default to be None. So, we should make this arg default to be None as well. 

I am not 100% sure if I understood your second question, but that code should be unrelated to this PR since that's from the scripts.py (if you are asking about the behavior, this PR will make `ray.init` work in the same way as `ray start`).

Reply:
It's not from scripts.py, it's the original 0.8.7 code in worker.py: 
![image](https://user-images.githubusercontent.com/14922/93385204-bf448b80-f81a-11ea-8445-afde8fd6fac6.png)

I want to make sure that we are preserving the previous behaviour of node ip address. You should base your PR on the 0.8.7 code, not whatever is currently in master which is broken.

Reply:
Ah, it makes a total sense. I will make sure the behavior is the same and ping you again.